### PR TITLE
corrected cmake scripts to find existing PETSC and SLEPC

### PR DESCRIPTION
**Pull request by loylick**
This is actually not a regular pull request, because its applying would  be irrelevant for other OS than windows. However the proposed modifications are exactly what helps cmake on windows to pick up installed PETSC and SLEPC packages. Some more remarks about building system. 
1) Finding external packages seems not to work at all under cygwin shell because of /cygdrive prefix in all paths. So it is advisable to use minGW shell to configure Dealii with external packages.
2)  Since external packages like PETSC and SLEPC use cygwin shell for configuring, dealii picks up some include paths with /cygdrive prefix and afterwards while compiling it does not find some necessary include files. In the following pic you can see /I"\cygdrive\h\dev\petsc-3.7.5\include\petsc\mpiuni" ( PETSC uses mpiuni/mpi.h for building with -with-mpi=0 flag, this mpi.h from PETSC package one should use to compile Dealii too) path in generated by cmake Visual Studio projet. I had to manually modify it in order to make this path visible for VS. Paths  to SLEPC and PETSC directories were pointed correctly nevertheless.
cygdrive_include
3) The same thing for examples which use PETSC or SLEPC libraries. In generated VS project for examples I had to point manually not only the include path but also add libraries libf2cblas.lib libf2clapack.lib that were built by PETSC and required by SLEPC. The libraries reside in the same directory as libpetsc.lib.  
A detailed discription of Dealii+PETSC+SLEPC installation procedure you can find at:
https://github.com/dealii/dealii/issues/3060

### CODE REVIEW

File: cmake/modules/FindPETSC.cmake [hunk 1]
```diff
@@ -38,11 +38,12 @@ SET_IF_EMPTY(PETSC_DIR "$ENV{PETSC_DIR}")
 SET_IF_EMPTY(PETSC_ARCH "$ENV{PETSC_ARCH}")
 
 DEAL_II_FIND_LIBRARY(PETSC_LIBRARY
-  NAMES petsc
+  NAMES libpetsc
```
Comment (tamiko) [inline cmake/modules/FindPETSC.cmake]: `NAMES` declares a list of one ore more alternative names. So change this to `NAMES petsc libpetsc` please.

File: cmake/modules/FindPETSC.cmake [hunk 2]
```diff
@@ -38,11 +38,12 @@ SET_IF_EMPTY(PETSC_DIR "$ENV{PETSC_DIR}")
...
   HINTS ${PETSC_DIR} ${PETSC_DIR}/${PETSC_ARCH}
   PATH_SUFFIXES lib${LIB_SUFFIX} lib64 lib
   )
 
+MESSAGE(STATUS "LIB_SUFFIX=\"${LIB_SUFFIX}\"")
```
Comment (tamiko) [inline cmake/modules/FindPETSC.cmake]: Please remove this stray message call.

File: cmake/modules/FindSLEPC.cmake [hunk 3]
```diff
@@ -39,7 +39,7 @@ SET_IF_EMPTY(PETSC_ARCH "$ENV{PETSC_ARCH}")
 #
 
 DEAL_II_FIND_LIBRARY(SLEPC_LIBRARY
-  NAMES slepc
+  NAMES libslepc
```
Comment (tamiko) [inline cmake/modules/FindSLEPC.cmake]: As above, `NAMES slepc libslepc`

**Comment by tamiko** [comment]
@loylick Can you please rebase and squash to one commit?

**Comment by loylick** [comment]
@tamiko I do not know how to rebase.

**Comment by tamiko** [comment]
@loylick No problem. I will merge by hand and do that for you.

**Pull request closed without merging**